runners/samza/src/main/java/org/apache/beam/runners/samza/runtime/SamzaTimerInternalsFactory.java

```diff
@@ -193,7 +201,7 @@ public Collection<KeyedTimerData<K>> removeReadyTimers() {
     }
     LOG.debug("Removed {} ready timers", readyTimers.size());
 
-    if (readyTimers.size() == maxReadyTimersToProcessOnce
+    if (readyTimers.size() >= maxReadyTimersToProcessOnce
```

Comment:
This change may not be necessary since readyTimers size is incremented by one per loop.